### PR TITLE
feat: add posting to realtime extensions

### PR DESCRIPTION
**Pull request by karolsojko**
@mobitar as to not make this PR bigger - for review purposes

I've simplified to just posting to realtime extensions. The following features will be added in consecutive PRs:
- posting to backup extensions
- sending failed backup emails
- creating temp S3 file for the extensions to pickup

Additionally I've replaced fetching user key params here from the Auth Service instead of the database (slowly getting rid of User domain in Syncing Server)

### CODE REVIEW

File: src/Domain/Item/ContentType.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export enum ContentType {
   Note = 'Note',
-  ItemsKey = 'SN|ItemsKey'
+  ItemsKey = 'SN|ItemsKey',
+  Extension = 'SF|Extension',
```
Comment (atmoio) [inline src/Domain/Item/ContentType.ts]:
```suggestion
  ServerExtension = 'SF|Extension',
```

This is what I would have called it earlier on if I understood what it was to become. ServerExtensions will likely be deprecated after we launch UserSettings, which will be used for cloud backup settings instead.

File: src/Infra/HTTP/AuthHttpService.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export class AuthHttpService implements AuthHttpServiceInterface {
+  constructor (
+    @inject(TYPES.HTTPClient) private httpClient: SuperAgentStatic,
+    @inject(TYPES.AUTH_SERVER_URL) private authServerUrl: string,
+  ) {
+  }
+
+  async getUserKeyParams(email: string, authenticated: boolean): Promise<KeyParams> {
+    const keyParamsResponse = await this.httpClient
+      .get(`${this.authServerUrl}/users/params`)
```
Comment (atmoio) [inline src/Infra/HTTP/AuthHttpService.ts:17]: So I'm not sure how this works with the new server, but with the Ruby server, getting auth params for a user requires their 2FA token. Is this the same here? You might be able to make an exception for internal requests.

Comment (karolsojko) [reply src/Infra/HTTP/AuthHttpService.ts:17]: This service basically calls Auth Server to execute the code which was initially in this project.

Code in auth: https://github.com/standardnotes/auth/blob/develop/src/Controller/UsersController.ts#L91-L108

**Review by atmoio — APPROVED** [review]
Only concern is the 2FA required for getting auth params, other than that looks good to me.

File: src/Infra/HTTP/AuthHttpService.ts (hunk 2 above)
Comment (atmoio) [reply src/Infra/HTTP/AuthHttpService.ts:17]: Hmm but in that code snippet I'm not seeing 2FA mention. We have this in SS Ruby: https://github.com/standardnotes/syncing-server/blob/develop/app/controllers/api/auth_controller.rb#L238

**Comment by karolsojko** [comment]
> Only concern is the 2FA required for getting auth params, other than that looks good to me.

MFA is still in place: https://github.com/standardnotes/syncing-server-js/pull/46/files#diff-2de915c7a5fd8e11de4881d1483e3fe6996530fd4d99a603f0c4a1e264c0002aR35-R72

It's just strictly generating the key params that was delegate to Auth ... so as the syncing-server-js knows less and less about the `User`.

the generating of key params is used in `syncing-server-js` in 2 places ... in the AuthController and in syncing-server-js-worker `ItemsSyncedEventHandler` (`extension_job.rb` in legacy syncing-server).

**Pull request merged**